drivers/ethernet/eth_mcux.c

```diff
@@ -394,19 +396,37 @@ static bool eth_get_ptp_data(struct net_if *iface, struct net_pkt *pkt,
 		if (ntohs(hdr_vlan->type) != NET_ETH_PTYPE_PTP) {
 			return false;
 		}
+
+		eth_hlen = sizeof(struct net_eth_vlan_hdr);
 	} else
 #endif
 	{
 		if (ntohs(NET_ETH_HDR(pkt)->type) != NET_ETH_PTYPE_PTP) {
 			return false;
 		}
+
+		eth_hlen = sizeof(struct net_eth_hdr);
 	}
 
 	net_pkt_set_priority(pkt, NET_PRIORITY_CA);
 
 	if (ptpTsData) {
 		/* Cannot use GPTP_HDR as net_pkt fields are not all filled */
-		struct gptp_hdr *hdr = gptp_get_hdr(pkt);
+		struct gptp_hdr *hdr;
+
+		/* In TX, the first net_buf contains the Ethernet header
+		 * and the actual gPTP header is in the second net_buf.
+		 * In RX, the Ethernet header + other headers are in the
+		 * first net_buf.
+		 */
+		if (is_tx == true) {
```

Comment:
Well, that's too much IMHO ;-). Boolean values should be tested as `(booly)` and `(!booly)`.

Reply:
I agree, but wasn't it some misra rule that says we need to do full check here?

Reply:
To the best of my knowledge, MISRA requires that boolean expressions are used in if, while, etc. conditions. A boolean variable itself is a boolean expression. Calling to @ceolin for clarification.

Reply:
(Otherwise, I grepped the codebase, and we have that pattern crept in already...)

Reply:
@pfalcon you're right, a boolean variable is a boolean expression. No need explicit comparison.